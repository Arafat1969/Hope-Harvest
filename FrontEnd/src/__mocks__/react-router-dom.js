// src/__mocks__/react-router-dom.js
const React = require("react");

function makeAnchor({ children, to, className, ...rest }) {
  // ensure className is a string
  const cn = Array.isArray(className) ? className.join(" ") : className;
  return (
    <a href={to} className={cn} {...rest}>
      {children}
    </a>
  );
}

module.exports = {
  BrowserRouter: ({ children }) => <>{children}</>,
  MemoryRouter: ({ children }) => <>{children}</>,
  Routes: ({ children }) => <>{children}</>,
  Route: ({ element }) => element,

  Link: (props) => makeAnchor(props),
  NavLink: (props) => makeAnchor(props),

  useNavigate: () => () => {},
  useLocation: () => ({ pathname: "/" }),
};
