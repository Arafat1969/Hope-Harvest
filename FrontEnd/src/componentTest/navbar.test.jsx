import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Navbar from "../components/navbar";

const mockLogout = jest.fn();
const renderNavbar = (props = {}) => {
  const defaultProps = {
    isAuthenticated: false,
    user: null,
    onLogout: mockLogout,
  };
  return render(
    <MemoryRouter>
      <Navbar {...defaultProps} {...props} />
    </MemoryRouter>
  );
};

describe("Navbar Component", () => {
  beforeEach(() => {
    mockLogout.mockClear();
  });

  it("shows login button when not authenticated", () => {
    renderNavbar();
    expect(screen.getByText("Login")).toBeInTheDocument();
    expect(screen.queryByText("Dashboard")).not.toBeInTheDocument();
  });

  it("shows dashboard and profile when authenticated as user", () => {
    renderNavbar({ isAuthenticated: true, user: { role: "USER" } });
    expect(screen.getByText("Dashboard")).toBeInTheDocument();
    expect(screen.getByText("Profile")).toBeInTheDocument();
  });

  it("shows admin links when authenticated as admin", () => {
    renderNavbar({ isAuthenticated: true, user: { role: "ADMIN" } });
    expect(screen.getByText("Admin Dashboard")).toBeInTheDocument();
    expect(screen.getByText("Manage Users")).toBeInTheDocument();
  });

  it("calls logout handler when logout clicked", () => {
    renderNavbar({ isAuthenticated: true, user: { role: "USER" } });
    fireEvent.click(screen.getByText("Logout"));
    expect(mockLogout).toHaveBeenCalled();
  });

  it("matches snapshot for guest user", () => {
    const { asFragment } = renderNavbar();
    expect(asFragment()).toMatchSnapshot();
  });
});