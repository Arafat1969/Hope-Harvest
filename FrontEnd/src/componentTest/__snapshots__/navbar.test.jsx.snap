// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component matches snapshot for guest user 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="container-fluid py-2 border-bottom topRow"
    >
      <div
        class="d-flex justify-content-between align-items-center"
      >
        <a
          class="logoText"
          href="/"
        >
          Â HOPE HARVEST
        </a>
        <div
          class="d-flex align-items-center gap-2"
        >
          <button
            class="btn button buttonContact"
          >
            Contact Us
          </button>
          <a
            class="btn text-white button buttonLogin"
            href="/login"
          >
            Login
          </a>
        </div>
      </div>
    </div>
    <nav
      class="navbar navbar-expand-md navbar-dark bottomNav transparentNav"
    >
      <div
        class="container-fluid"
      >
        <button
          aria-controls="navbarNav"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-bs-target="#navbarNav"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon navbarTogglerIcon"
          />
        </button>
        <div
          class="collapse navbar-collapse justify-content-center"
          id="navbarNav"
        >
          <ul
            class="navbar-nav navLinksContainer"
          >
            <li
              class="nav-item"
            >
              <a
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                href="/about"
              >
                About
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                href="/campaigns"
              >
                Campaigns
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                href="/donate"
              >
                Donate
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                href="/track-donation"
              >
                Track Donation
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                href="/volunteer-activity"
              >
                Volunteer
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                href="/apply-for-funds"
              >
                Apply for Funds
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                href="/gallery"
              >
                Gallery
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;
